### DISCUSSION TITLE
Expected behavior upon JWT verification failure

### DISCUSSION THREAD

**Discussion by hojoungjang** [panva/jose#735] [Q&A] [answered]
I am not sure what should happen when the verification using `jose.jwtVerify(...)` fails (i.e. the given JWT is invalid).
Can you please provide some examples? Do we simply catch errors like this?

```ts
const jwtPubKey = "..."
const key = await jose.importSPKI(jwtPubKey, "RS256");
try {
  const { payload, protectedHeader } = await jose.jwtVerify(token, key);
} catch (error) {
  console.log(error);
}
```

For my use case, I need to determine when the verification fails.

I assume there are different scenarios for verification failures (ex. expired JWT, signature mismatch and so on).
Will catching the error cover all cases where the JWT is invalid?

**Comment by panva** [comment, marked as answer]
The returned promise will **only** resolve when everything's in order. In all other cases the promise will reject.

**Reply by hojoungjang** [reply]
Got it, thanks!
